File: Frontend/edu-quiz/src/app/models/filled-quiz.model.ts
import { Question } from "./question.model";

export class FilledQuiz {
    constructor (
        public id: number,
        public userId: number,
        public quizId: number,
        public quizName: string,
        public quizCreatorId: number,
        public isChecked: boolean,
        public questions: Question[]
    ) {}
}

File: Backend/EduQuizWebAPI/Models/FilledQuizModel.cs
using EduQuizDBAccess.Entities;

namespace EduQuizWebAPI.Models {
    public class FilledQuizModel {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int QuizId { get; set; }
        public string QuizName { get; set; } = null!;
        public int QuizCreatorId { get; set; }
        public bool IsChecked { get; set; }
        public ICollection<QuestionModel>? Questions { get; set; }
    }
}
